app/controllers/foreman_inventory_upload/uploads_controller.rb

```diff
@@ -24,6 +24,21 @@ def throw_flash_error(message)
       )
     end
 
+    def enable_cloud_connector
+      cloud_connector = ForemanRhCloud::CloudConnector.new
+      cloud_connector.install
+    end
+
+    def cloud_connector_status
+      cloud_connector = ForemanRhCloud::CloudConnector.new
+
+      if (job = cloud_connector.running_job)
```

Comment:
Why can't we always return the job? Even if it's running, we have the link to the running job and not the task. I'm probably missing something. But `running_job` seems to either return `nil` or the job itself. Why do we load task from it if it's job already?

Reply:
I was not sure how the UI would know from the returned data here that the job was still executing. Are you saying return the job and have the UI then query the Jobs API for status?

Reply:
I think I'm saying return the job and the job status in the custom json response. User shouldn't be punished by our insufficient APIs, we should give them a link to the job instead of a task and their browser shouldn't need to do 2 queries for that in my opinion.

Reply:
Does this makes sense?
https://github.com/theforeman/foreman_rh_cloud/blob/80e08cd4a3ed2bace536e4acc4343c983f0667aa/app/controllers/foreman_inventory_upload/uploads_controller.rb#L32-L35
https://github.com/theforeman/foreman_rh_cloud/blob/80e08cd4a3ed2bace536e4acc4343c983f0667aa/webpack/ForemanInventoryUpload/Components/PageHeader/components/CloudConnectorButton/CloudConnectorSelectors.js#L8-L12
from my PR https://github.com/theforeman/foreman_rh_cloud/pull/458

Reply:
yeah, I definitely mind the implementation less, e.g. we can use task and we can do extra API query to fetch the job, it just feels suboptimal since we already have all of that info on the server side